hubmap-auth/src/app.py

```diff
@@ -284,6 +284,47 @@ def file_auth():
     return response_401
 
 
+@app.route('/umls_auth', methods = ['GET'])
+def umls_auth():
+    logger.info("======umls_auth Original request.headers======")
+    logger.info(request.headers)
+
+    # Nginx auth_request only cares about the response status code
+    # it ignores the response body
+    # We use body here only for description purposes and direct visit to this endpoint
+    response_200 = make_response(jsonify({"message": "OK: Authorized"}), 200)
+    response_401 = make_response(jsonify({"message": "ERROR: Unauthorized"}), 401)
+    response_403 = make_response(jsonify({"message": "ERROR: Forbidden"}), 403)
+    response_500 = make_response(jsonify({"message": "ERROR: Internal Server Error"}), 500)
+
+    # Note: 400 and 404 are not supported http://nginx.org/en/docs/http/ngx_http_auth_request_module.html
+    # Any response code other than 200/401/403 returned by the subrequest is considered an error 500
+    # The end user or client will never see 404 but 500
+    response_400 = make_response(jsonify({"message": "ERROR: Bad Request"}), 400)
+    response_404 = make_response(jsonify({"message": "ERROR: Not Found"}), 404)
+
+    orig_uri = None
+
+    if "X-Original-URI" in request.headers:
+        orig_uri = request.headers.get("X-Original-URI")
+
+    if orig_uri is None:
+        return response_400
```

Comment:
As long as the sub-request gets to this `/umls_auth` endpoint, this `orig_uri` won't be `None`. So no need to check here, plus returning 400 will become 500 and cause confusions.